akka-docs/rst/general/stream/stages-overview.rst

```diff
@@ -1,21 +1,373 @@
 .. _stages-overview:
 
-###############################################
 Overview of built-in stages and their semantics
-###############################################
+===============================================
 
-All stages by default backpressure if the computation they encapsulate is not fast enough to keep up with the rate of
+
+Source stages
+-------------
+These built-in sources are available from `akka.stream.scaladsl.Source` and `akka.stream.javadsl.Source`
+
+
+
+fromIterator
+^^^^^^^^^^^^
+Streams the values from an iterator, requesting the next value when there is demand. If the iterator
+performs blocking operations, make sure to run it on a separate dispatcher.
+
+*emits* when the next value returned from the iterator
+*completes* when the iterator reaches it's end
+
+single
+^^^^^^
+Streams a single object
+
+*emits* the value once
+*completes* when the single value has been emitted
+
+repeat
+^^^^^^
+Streams a single object repeatedly
+
+*emits* the same value repeatedly when there is demand
+*completes* never
+
+tick
+^^^^
+A periodical repeated stream of an arbitrary object. Delay of first tick is specified
+separately from interval of the following ticks.
+
+*emits* periodically, if there is downstream backpressure ticks are skipped
+*completes* never
+
+fromFuture
+^^^^^^^^^^
+The value of the future is sent when the future completes and there is demand.
+If the future fails the stream is failed with that exception.
+
+*emits* the future completes
+*completes* after the future has completed
+
+unfold
+^^^^^^
+Streams the result of a function as long as it returns a ``Some`` or non-empty ``Optional``, the value inside the optional
+consists of a tuple (or ``Pair``) where the first value is a state passed back into the next call to the function allowing
+to pass a state. The first invocation of the provided fold function will receive the ``zero`` state.
+
+Can be used to implement many stateful sources without having to touch the more low level ``GraphStage`` API.
+
+*emits* when there is demand and the unfold function over the previous state returns non empty value
+*completes* when the unfold function returns an empty value
+
+unfoldAsync
```

Comment:
oh, did I convert that to CompletionStage as well? perhaps it is still missing …